src/main/java/lotto/LottoGame.java

```diff
@@ -0,0 +1,93 @@
+package lotto;
+
+import lotto.exception.GameException;
+import lotto.io.LottoIOHandler;
+import lotto.model.*;
+import lotto.provider.NumbersProvider;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static lotto.io.request.WinningNumberRequest.SPLIT_DELIMITER;
+
+public class LottoGame {
+
+    private final LottoIOHandler ioHandler = new LottoIOHandler();
+    private final NumbersProvider numbersProvider;
+
+    public LottoGame(NumbersProvider numbersProvider) {
+        this.numbersProvider = numbersProvider;
+    }
+
+    public void run() {
+        List<Lotto> lottos = purchaseLotto();
+        ioHandler.showPurchasedLottos(lottos);
+
+        Lotto normalNumbersOfLotto = createWinningNumbers();
+        WinningLotto winningLotto = createBonusNumber(normalNumbersOfLotto);
+
+        WinningResult winningResult = calculateResult(lottos, winningLotto);
+        showResult(winningResult);
+    }
+
+    private List<Lotto> purchaseLotto() {
+        try {
+            int budget = ioHandler.askPurchaseAmount();
+            LottoVendingMachine vendingMachine = new LottoVendingMachine(budget);
+            return vendingMachine.purchaseAll(numbersProvider);
+        } catch (GameException e) {
+            System.out.println(e.getMessage());
+            return purchaseLotto();
+        }
+    }
+
+    private Lotto createWinningNumbers() {
+        try {
+            String candidateWinningNumbers = ioHandler.askWinningNumbers();
+            return convertLottoFrom(candidateWinningNumbers);
+        } catch (GameException e) {
+            System.out.println(e.getMessage());
+            return createWinningNumbers();
+        }
+    }
+
+    private Lotto convertLottoFrom(String candidateWinningNumbers) {
+        List<Integer> winningNumbers = Arrays.stream(candidateWinningNumbers.split(SPLIT_DELIMITER))
+            .map(Integer::parseInt)
+            .toList();
```

Comment:
```suggestion
        List<Integer> winningNumbers = Stream.of(candidateWinningNumbers.split(SPLIT_DELIMITER)))
                .map(Integer::parseInt)
                .toList();
```

민혁님, 혹시 아래와 같이 스트림을 사용하는 방식에 대해 어떻게 생각하시나요? 
제안드리는 건 아니고, 단순히 의견이 궁금해서 여쭤봅니다. 스트림을 작성하다 보니 궁금해졌어요!

Reply:
제가 스트림이 익숙하지가 않아서.. 오히려 아래형식이 더 나아보일수도있다고 생각은하네요 ! 스트림으로 만들어준다는게 더 명시적일 수도 있다고 생각해요 !